lib/gateways/ssm-param-gateway.ts

```diff
@@ -0,0 +1,156 @@
+import * as log from 'lambda-log';
+import * as ssm from '@aws-sdk/client-ssm';
+import * as retryMiddleware from '@aws-sdk/middleware-retry'
+import { Provider } from '@aws-sdk/types';
+
+const SSM_RETRY_ATTEMPTS = 'SSM_RETRY_ATTEMPTS';
+const SSM_RETRY_ATTEMPTS_DEFAULT = 10;
+export class SsmGateway {
+
+    ssmClient: ssm.SSMClient;
+
+    constructor(ssmClient?: ssm.SSMClient) {
+
+        if (!ssmClient) {
+
+            const retryAttemptsEnv = process.env[SSM_RETRY_ATTEMPTS]
+
+            const retryAttempts = (retryAttemptsEnv) ? parseInt(retryAttemptsEnv) : SSM_RETRY_ATTEMPTS_DEFAULT;
+            const maxAttemptsProvider: Provider<number> = () => Promise.resolve(retryAttempts);
+
+            const adaptiveRetry = new retryMiddleware.AdaptiveRetryStrategy(maxAttemptsProvider, {
+            });
+            this.ssmClient = new ssm.SSMClient({
+                retryStrategy: adaptiveRetry,
+            })
+        }
+        if (process.env.LOG_LEVEL && process.env.LOG_LEVEL.toLocaleLowerCase() === 'debug') {
+            log.options.debug = true;
+        }
+    }
+
+    public async putParameter(name: string, value: string): Promise<void> {
+        log.debug(`api: aws ssm put-parameter`, { name, value })
+
+        let result: ssm.PutParameterCommandOutput;
+        try {
+
+            result = await this.ssmClient.send(
+                new ssm.PutParameterCommand({
+                    Name: `${name}`,
+                    Value: `${value}`,
+                    Overwrite: true,
+                    Type: ssm.ParameterType.STRING,
+                })
+            )
+
+            if (!result.Version) throw Error(`Failed Updating parameter: ${value}`);
```

Comment:
will this be able to tell if an existing parameter was updated? Version could be still the old version, right?

Reply:
Intent is just to check to ensure a parameter was created.   With the `--overwrite` option it should always create, unless it hits a retry limit  due to throttling - in which case another exception is thrown. 

I could probably just check to ensure there is a result.   Or maybe even not have this check.